Kinvey.Tests/DataStoreTests/DataStoreAutoIntegrationTests.cs

```diff
@@ -5627,189 +5789,261 @@ public async Task TestSaveAsync()
-                new ToDo { Name = "Name2", Details = "Details2", Value = 2 }
+                Name = "todo save",
+                Details = "details for save",
+                DueDate = "2016-04-22T19:56:00.961Z"
             };
 
+            await User.LoginAsync(TestSetup.user, TestSetup.pass, kinveyClient);
+
             // Act
-            var savedToDos = await autoToDoStore.SaveAsync(toDos);
+            var savedItem = await autoStore.SaveAsync(newItem);
 
-            var pendingWriteActions = kinveyClient.CacheManager.GetSyncQueue(toDosCollection).GetAll();
-            var existingToDos = await syncToDoStore.FindAsync();
+            var existingItemCache = await syncStore.FindByIDAsync(savedItem.ID);
+            var existingItemNetwork = await networkStore.FindByIDAsync(savedItem.ID);
+            
 
-            // Teardown
-            await autoToDoStore.RemoveAsync(savedToDos.Entities[0].ID);
-            await autoToDoStore.RemoveAsync(savedToDos.Entities[1].ID);
+            //Teardown
+            await networkStore.RemoveAsync(savedItem.ID);
 
             // Assert
-            Assert.AreEqual(2, savedToDos.Entities.Count);
-            Assert.AreEqual(0, savedToDos.Errors.Count);
-            Assert.AreEqual(toDos[0].Name, savedToDos.Entities[0].Name);
-            Assert.AreEqual(toDos[0].Details, savedToDos.Entities[0].Details);
-            Assert.AreEqual(toDos[0].Value, savedToDos.Entities[0].Value);
-            Assert.AreEqual(user.Id, savedToDos.Entities[0].Acl.Creator);
-            Assert.AreEqual(toDos[1].Name, savedToDos.Entities[1].Name);
-            Assert.AreEqual(toDos[1].Details, savedToDos.Entities[1].Details);
-            Assert.AreEqual(toDos[1].Value, savedToDos.Entities[1].Value);
-            Assert.AreEqual(user.Id, savedToDos.Entities[1].Acl.Creator);
-            Assert.AreEqual(0, pendingWriteActions.Count);
-            Assert.AreEqual(2, existingToDos.Count);
-            Assert.IsNotNull(existingToDos.FirstOrDefault(e=> e.ID == savedToDos.Entities[0].ID));
-            Assert.IsNotNull(existingToDos.FirstOrDefault(e => e.ID == savedToDos.Entities[1].ID));
+            Assert.IsNotNull(savedItem);
+            Assert.AreEqual(newItem.Name, savedItem.Name);
+            Assert.AreEqual(newItem.Details, savedItem.Details);
+            Assert.AreEqual(newItem.DueDate, savedItem.DueDate);
+            Assert.IsNotNull(existingItemCache);
+            Assert.AreEqual(newItem.Name, existingItemCache.Name);
+            Assert.AreEqual(newItem.Details, existingItemCache.Details);
+            Assert.AreEqual(newItem.DueDate, existingItemCache.DueDate);
+            Assert.IsNotNull(existingItemCache.Acl);
+            Assert.IsNotNull(existingItemCache.Kmd);
+            Assert.AreNotEqual(string.Empty, existingItemCache.Kmd.entityCreationTime);
+            Assert.AreNotEqual(string.Empty, existingItemCache.Kmd.lastModifiedTime);
```

Comment:
Yes, it would. I agree with you.
If the item is initially save as local and not overridden ect and lmt, Acl and Kmd will be null and ect and lmt will not exist at all.